src/engine/strat_engine/dmdevice.rs

```diff
@@ -74,3 +79,39 @@ pub fn format_thinpool_name(pool_uuid: &PoolUuid, role: ThinPoolRole) -> String
                    pool_uuid.simple().to_string(),
                    role);
 }
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct SThinDevId {
+    value: u32,
+}
+
+impl fmt::Display for SThinDevId {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Display::fmt(&self.value, f)
+    }
+}
+
+impl ThinDevId for SThinDevId {}
+
+impl SThinDevId {
+    /// Instantiate a new, random, thindev id.
+    pub fn new_random() -> SThinDevId {
+        SThinDevId { value: rand::random::<u32>() >> 8 }
+    }
+}
```

Comment:
I was thinking it might be better for Stratis to allocate these sequentially... but I suppose it is a toss up between that and a random number.